### PR TITLE
Migrate to native classes

### PR DESCRIPTION
**Pull request by josemarluedke**
This moves the addon code to native classes and changes how to setup clientOptions, link and cache. 

**This drops support for Ember v2.18**. Minimum tested version is Ember v3.4 using `ember-native-class-polyfill`.

### CODE REVIEW

**Comment by josemarluedke** [comment]
@bgentry I believe this is ready for a review. Changes seem to work as expected, but it does removes support for Ember 2.18 as tests kept failing, making the minimum Ember tested version 3.4. 

@buschtoens Mind taking a look in this PR as well? Maybe @pzuraq could also have his eyes on this?

**Comment by josemarluedke** [comment]
I actually refactored how we do the setup of `clientOptions`, `link` and `cache`, removing the need for computed entirely. This does make things simper IMO and removes the need for requiring ember-decorators in runtime. I kept `@ember-decorators/babel-transforms` as it enables using native classes, class fields and whatnot. Will research if we really need to keep it long term.

File: addon/apollo/query-manager.js [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
-import { alias } from '@ember/object/computed';
-import { A } from '@ember/array';
+export default class QueryManager {
+  apollo = undefined;
+  activeSubscriptions = [];
 
-export default EmberObject.extend({
-  apollo: service(),
-  apolloClient: alias('apollo.client'),
+  constructor(apoloService) {
```
Comment (christophermlne) [inline addon/apollo/query-manager.js]: typo

**Comment by josemarluedke** [comment]
@bgentry I believe this PR can move forward. Mind doing a review?

**Comment by josemarluedke** [comment]
@bgentry I have refactor this a little bit to remove the breaking change, instead I have deprecated the old way of overwriting apollo options.

**Comment by bgentry** [comment]
Released in v2.0.0-beta.4

**Pull request merged**